### PR TITLE
(feat) support max connection setting for database configuration

### LINKED ISSUE #7427: [Feature request] Allow max db connection pool configuration
Currently, it's not possible to configure max pool size (SetMaxOpenConns) which causes problems when having many Grafana instances.

### PR DESCRIPTION
**Pull request by huydx**
Fix https://github.com/grafana/grafana/issues/7427

### CODE REVIEW

File: pkg/services/sqlstore/sqlstore.go [hunk 1]
```diff
@@ -177,6 +198,18 @@ func LoadConfig() {
 		DbCfg.Host = sec.Key("host").String()
 		DbCfg.Name = sec.Key("name").String()
 		DbCfg.User = sec.Key("user").String()
+		DbCfg.MaxConn, err = sec.Key("max_conn").Int()
```
Comment (bergquist) [inline pkg/services/sqlstore/sqlstore.go]: Please use `.MustInt(0)` instead. Reduce the amount of code needed to read the setting. 0 is the default value anyway :)

File: pkg/services/sqlstore/sqlstore.go [hunk 2]
```diff
@@ -150,11 +153,29 @@ func getEngine() (*xorm.Engine, error) {
 	}
 
 	sqlog.Info("Initializing DB", "dbtype", DbCfg.Type)
-	return xorm.NewEngine(DbCfg.Type, cnnstr)
+	engine, err := xorm.NewEngine(DbCfg.Type, cnnstr)
+	if err != nil {
+		return nil, err
+	} else {
+		if DbCfg.MaxConn != 0 {
```
Comment (bergquist) [inline pkg/services/sqlstore/sqlstore.go]: This if statement is kinda of redundant. xorm will ignore any connection setting that is 0. So just pass in the DbCfg.MaxConn value. Same for `MaxOpenConn` and `MaxIdleConn`

**Review by bergquist — CHANGES_REQUESTED** [review]
Great work! But this code could be simplify a bit.

File: pkg/services/sqlstore/sqlstore.go (hunk 2 above)
Comment (huydx) [inline pkg/services/sqlstore/sqlstore.go]: oh, didn't know that, thanks for the information!

**Comment by huydx** [comment]
fixed

**Review by bergquist — APPROVED** [review]

**Comment by bergquist** [comment]
Merged!

**Pull request closed without merging**